From ghaskins@novell.com Mon Mar 24 17:40:01 2008
Date: Fri, 07 Mar 2008 07:11:43 -0500
From: Gregory Haskins <ghaskins@novell.com>
To: rostedt@goodmis.org, mingo@elte.he
Cc: linux-kernel@vger.kernel.org, linux-rt-users@vger.kernel.org,
     ghaskins@novell.com
Subject: [PATCH] RESEND: fix cpus_allowed settings

    [ The following text is in the "utf-8" character set. ]
    [ Your display is set for the "iso-8859-1" character set.  ]
    [ Some characters may be displayed incorrectly. ]

Hi Ingo, Steve,
  I sent this patch a few weeks ago along with the migration_disable
series.  I think the controversy with the migration_disable feature
may have resulted in this fix being overlooked.  This patch is against
-rt, but the bug theoretically affects both -rt and sched-devel/mainline.
I can also whip up a sched-devel based patch if you like, but I think it
will apply trivially to both places.

Please consider it for inclusion.

Regards,
-Greg

-------------------------------
Subject: fix cpus_allowed settings

We miss setting nr_cpus_allowed for the kthread case since the normal
set_cpus_allowed() function is not used.

Signed-off-by: Gregory Haskins <ghaskins@novell.com>
---

 kernel/kthread.c |    1 +
 1 file changed, 1 insertion(+)

Index: linux-2.6.25.4-rt1/kernel/kthread.c
===================================================================
--- linux-2.6.25.4-rt1.orig/kernel/kthread.c	2008-05-17 08:26:04.000000000 -0400
+++ linux-2.6.25.4-rt1/kernel/kthread.c	2008-05-17 08:28:11.000000000 -0400
@@ -180,6 +180,7 @@ void kthread_bind(struct task_struct *k,
 	wait_task_inactive(k);
 	set_task_cpu(k, cpu);
 	k->cpus_allowed = cpumask_of_cpu(cpu);
+	k->rt.nr_cpus_allowed = 1;
 }
 EXPORT_SYMBOL(kthread_bind);
 
